### PR TITLE
RM/HM/LIE-HM interpolated pressure

### PR DESCRIPTION
**Pull request by endJunction**
Move LIE/HM interpolate_pressure to `NumLib::interpolateToHigherOrderNodes`.

Use in RichardsMechanics and HydroMechanics processes.

### CODE REVIEW

File: NumLib/Function/Interpolation.h [hunk 1]
```diff
@@ -85,4 +88,52 @@ void shapeFunctionInterpolate(
                                         interpolated_values...);
 }
 
+/// Interpolates \c node_values given in lower order element nodes (e.g. the
+/// base nodes) to higher order element's nodes (e.g. quadratic nodes) and
+/// writes the result into the global property vector.
+///
+/// The base nodes' values are copied.  For each higher order node the shape
+/// matrices are evaluated for the lower order element (the base nodes), and
+/// used for the the scalar quantity interpolation.
```
Comment (chleh) [inline NumLib/Function/Interpolation.h]: Maybe it should be mentioned that the node_values belong to a scalar field.

File: NumLib/Function/Interpolation.h [hunk 2]
```diff
@@ -85,4 +88,52 @@ void shapeFunctionInterpolate(
...
+
+    // Shape matrices storage reused in the interpolation loop.
+    ShapeMatrices shape_matrices{SF::DIM, GlobalDim, SF::NPOINTS};
+
+    // Interpolate values for higher order nodes.
+    for (int n = number_base_nodes; n < number_all_nodes; ++n)
+    {
+        // Evaluated at higher order nodes' coordinates.
+        fe.template computeShapeFunctions<ShapeMatrixType::N>(
+            NaturalCoordinates<HigherOrderMeshElementType>::coordinates[n]
```
Comment (chleh) [inline NumLib/Function/Interpolation.h]: The `HigherOrderMeshElementType` is the "true" type of `element`, isn't it? Maybe that could be commented
or asserted: `assert(dynamic_cast<HigherOrderMeshElementType*>(&element));`.

File: NumLib/Function/Interpolation.h [hunk 3]
```diff
@@ -85,4 +88,52 @@ void shapeFunctionInterpolate(
...
+                .data(),
+            shape_matrices, GlobalDim, is_axially_symmetric);
+
+        std::size_t const global_index = element.getNodeIndex(n);
```
Comment (chleh) [inline NumLib/Function/Interpolation.h]: Does that work with domain decomposition?

Comment (endJunction) [reply NumLib/Function/Interpolation.h]: I think so, because the node numbering is local for each partition, and so are the mesh properties...

**Pull request merged**